sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,558 @@
+ * remote.netif = gnrc_netif_iter(NULL)->pid;   // only if gnrc_netif_numoff == 1
+ * if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, addr_str)) {
+ *     puts("Error parsing destination address");
+ *     return;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After the UDP sock is created, we can proceed with creating the DTLS sock.
+ * Before sending the packet, we must first establish a session with the remote
+ * endpoint using @ref sock_dtls_establish_session(). If the handshake is
+ * successfull and the session is established, we can use its `remote` to send
+ * packets to it with @ref sock_dtls_send().
+ *
+ * If error during any of these operation, the session is terminated and the sock
+ * is closed using @ref sock_dtls_terminate_session() and
+ * @ref sock_dtls_close().
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = sock_dtls_create(&dtls_sock, &udp_sock, DTLS_SOCK_CLIENT_TAG, 0);
+ * if (res < 0) {
+ *     puts("Error creating DTLS sock");
+ *     return;
+ * }
+ *
+ * res = sock_dtls_establish_session(&dtls_sock, &remote, &session);
+ * if (res < 0) {
+ *     printf("Error establishing session: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_send(&dtls_sock, &session, data, datalen);
+ * if (res < 0) {
+ *     printf("Error sending DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT);
+ * if (res < 0) {
+ *     printf("Error receiving DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * end:
+ * puts("Terminating");
+ * sock_dtls_terminate_session(&dtls_sock, &session);
+ * sock_dtls_close(&dtls_sock);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @{
+ *
+ * @file
+ * @brief   DTLS sock definitions
+ *
+ * @author  Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#ifndef NET_SOCK_DTLS_H
+#define NET_SOCK_DTLS_H
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+#include "net/sock.h"
+#include "net/sock/udp.h"
+#include "net/credman.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Type for a DTLS sock object
+ *
+ * @note    API implementors: `struct sock_dtls` needs to be defined by
+ *          an implementation-specific `sock_dtls_types.h`.
+ */
+typedef struct sock_dtls sock_dtls_t;
+
+/**
+ * @brief Information about an established session with a remote endpoint.
+ */
+typedef struct sock_dtls_session sock_dtls_session_t;
+
+/**
+ * @brief Called exactly once during autoinit.
+ *
+ * Calls the initialization function required by the DTLS stack used.
+ */
+void sock_dtls_init(void);
+
+/**
+ * @brief Creates a new DTLS sock object
+ *
+ * @param[out] sock     The resulting DTLS sock object
+ * @param[in] udp_sock  Existing UDP sock to be used underneath
+ * @param[in] tag       Credential tag of the sock. The sock will only use
+ *                      credentials with the same tag given here.
```

Comment:
Maybe add a

```
@see net_credman
```

to this functions doc, so people know, what you talk about here ;-).

Reply:
Added link to `net_credman`.